.github/workflows/build.yml

```
@@ -0,0 +1,29 @@
+name: 'build'
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        idf_ver: ["v4.1", "v4.2", "v4.3"]
+        idf_target: ["esp32"]
+        include:
+          - idf_ver: "v4.2"
+            idf_target: esp32s2
+            # TODO: enable C3 build
+          #- idf_ver: "v4.3"
+          #  idf_target: esp32c3
+    runs-on: ubuntu-20.04
+    container: espressif/idf:release-${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v1

```

Checkout v2 does shallow checkouts by default, which makes it a lot faster, and probably reduces the bandwidth costs on GitHub's end as well. 

```suggestion
      - uses: actions/checkout@v2
```

---

The action fails with checkout@v2, will investigate further

---

It's [supposed to have `git`](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/tools/idf-docker-image.html), however, I wouldn't be surprised if the version is older than what `actions/checkout@v2` expects.

---

I'll revert to checkout@v1 for now